src/content/BatchItem.ts

```diff
@@ -0,0 +1,196 @@
+  }
+};
+
+/**
+ * Serializes the batch item
+ * @param writer
+ * @param batchRequestData
+ */
+export const serializeBatchItem = (
+  writer: SerializationWriter,
+  batchRequestData: Partial<BatchItem> | undefined | null = {},
+): void => {
+  if (batchRequestData) {
+    writer.writeStringValue("id", batchRequestData.id);
+    writer.writeStringValue("method", batchRequestData.method);
+    writer.writeStringValue("url", batchRequestData.url);
+    const headers: UntypedNode = {
+      getValue: (): unknown => batchRequestData.headers,
+      value: batchRequestData.headers,
+    };
+    writer.writeObjectValue("headers", headers);
+    const body: UntypedNode = {
+      getValue: (): unknown => batchRequestData.body,
+      value: batchRequestData.body,
+    };
+    writer.writeObjectValue("body", body);
+    writer.writeCollectionOfPrimitiveValues("dependsOn", batchRequestData.dependsOn);
+  }
+};
+
+/**
+ * BatchResponseCollection ParsableFactory
+ * @param _parseNode
+ */
+export const createBatchResponseContentFromDiscriminatorValue = (
+  _parseNode: ParseNode | undefined,
+): ((instance?: Parsable) => Record<string, (node: ParseNode) => void>) => {
+  return deserializeIntoBatchResponseContent;
+};
+
+/**
+ * Deserializes the batch response body
+ * @param batchResponseBody
+ */
+export const deserializeIntoBatchResponseContent = (
+  batchResponseBody: Partial<BatchResponseCollection> | undefined = {},
+): Record<string, (node: ParseNode) => void> => {
+  return {
+    responses: n => {
+      batchResponseBody.responses = n.getCollectionOfObjectValues(createBatchResponseFromDiscriminatorValue);
+    },
+  };
+};
+
+/**
+ * BatchItem ParsableFactory
+ * @param _parseNode
+ */
+export const createBatchResponseFromDiscriminatorValue = (
+  _parseNode: ParseNode | undefined,
+): ((instance?: Parsable) => Record<string, (node: ParseNode) => void>) => {
+  return deserializeIntoBatchResponse;
+};
+
+/**
+ * Deserializes the batch item
+ * @param batchResponse
+ */
+export const deserializeIntoBatchResponse = (
+  batchResponse: Partial<BatchResponse> | undefined = {},
+): Record<string, (node: ParseNode) => void> => {
+  return {
+    id: n => {
+      batchResponse.id = n.getStringValue();
+    },
+    headers: n => {
+      const headers: UntypedNode = n.getObjectValue<UntypedNode>(createUntypedNodeFromDiscriminatorValue);
+      batchResponse.headers = getUntypedNodeValue(headers) as Record<string, string> | null;
+    },
+    body: n => {
+      const body: UntypedNode = n.getObjectValue<UntypedNode>(createUntypedNodeFromDiscriminatorValue);
+      batchResponse.body = getUntypedNodeValue(body) as Record<string, unknown> | null;
+    },
+    status: n => {
+      batchResponse.status = n.getNumberValue();
+    },
+  };
+};
+
+/**
+ * @private
+ * Unwraps the untyped node value
+ * @param untypedValue
+ */
+const getUntypedNodeValue = (untypedValue: UntypedNode | null | undefined): unknown => {
```

Comment:
Ideally, this function should disappear. Its currently being used for 
- headers(which can possibly leverage `getObjectValue` to get a dictionary)
- the request body(which we should ideally hold a stream and then provide methods for the user to deserialize into the object they choose using a factory)

https://github.com/microsoftgraph/msgraph-sdk-design/blob/main/content/BatchResponseContent.md#usage-examples